src/intervals/wrapper_intervals.jl

```diff
@@ -0,0 +1,76 @@
+import Base: convert, promote_rule
+
+"""
+    SetInterval
+
+Interval type that works like a set.
+"""
+struct SetInterval{T}
+    interval::BaseInterval{T}
+end
+
+"""
+    RealSetInterval
+
+Interval type that works like a set, but is a subtype of `Real` nonetheless.
+"""
+struct RealSetInterval{T} <: Real
+    interval::BaseInterval{T}
+end
+
+"""
+    PseudoNumberInterval
+
+Interval type that works like a number and is a subtype of `Real`.
+
+Operation that returns `Bool` return `false` when the result can not be
+certified to be `true` for all elements in the interval.
+
+!!! warning
+    This behavior may have unexpected consequence, as for example for
+    `X = PseudoNumberInterval(-1..1)` both `X > 0` and `X <= 0` both return
+    `false`.
+"""
+struct PseudoNumberInterval{T} <: Real
+    interval::BaseInterval{T}
+end
+
+for IT in (:SetInterval, :RealSetInterval, :PseudoNumberInterval)
+    @eval ($IT)(params...) = ($IT)(BaseInterval(params...))
+    @eval convert(IT, x::BaseInterval) = ($IT)(x)
+    # @eval ($IT){T}(params...) = ($IT){T}(BaseInterval(params...))
+end
+
+AnyInterval{T, M} = Union{BaseInterval{T},
+                          SetInterval{T},
+                          RealSetInterval{T},
+                          PseudoNumberInterval{T}}
+
+# Alias for convenience.
+# Default is to the most restrictive version.
+Interval{T} = SetInterval{T}
+
+disambiguation_mode(val) = NoDisambiguation
+disambiguation_mode(::Union{SetInterval, RealSetInterval}) = ErrorOnAmbiguous
+disambiguation_mode(::PseudoNumberInterval) = FalseOnAmbiguous
```

Comment:
I still think the intervals should behave more like:
```suggestion
disambiguation_mode(::Union{SetInterval, RealSetInterval}) = IEEEConform
disambiguation_mode(::PseudoNumberInterval) = ErrorOnAmbiguous
```
(Here `IEEEConform` is the current behavior.) It's not quite clear to me if `FalseOnAmbiguous` would add anything since one could always wrap an `ErrorOnAmbiguous` type in a `try-catch` statement.

Reply:
The idea behind `FalseOnAmbiguous` was that it return `true` if and only if the tested predicate is guaranteed to be true, otherwise returning `false`. It turns out that for comparison, the IEEE standard (section 10.5.10) call the corresponding function `precedes` (and denotes it with a funky operator), while the `less` operator (denoted as `<` in the standard) represents something else.

Your are correct that this should simply refers to the standard and in the documentation it must be clear what we call `<`. It will be hard to be fully compliant with the notation of the standard in a meaningful way, because I don't know if it takes in account languages supporting unicode operators, nor the very Julian expectation of being able to throw new types into old algorithms.

Then I totally messed up the naming, the standard calling the described default interval as "set based", using the same convention we end up to your proposition (that I do not commit yet as `IEEEConform` is not defined).

Reply:
> Your are correct that this should simply refers to the standard and in the documentation it must be clear what we call `<`. It will be hard to be fully compliant with the notation of the standard in a meaningful way, because I don't know if it takes in account languages supporting unicode operators, nor the very Julian expectation of being able to throw new types into old algorithms.

I don't think that is an issue considering IEEE section 12.2 "Naming conventions for operations" states that implementations can use 'generic text names' like `add(a, b)` instead of `a+b`. The standard only specifies functionality. (In fact I find it rather unfortunate that they specified set operations using operators normally used for number operations, such as `<`.)